src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -1014,6 +1064,12 @@ func (r *runResult) addIndexBlockIfNotExists(
 }
 
 func (r *runResult) mergedResult(other *runResult) *runResult {
+	r.Lock()
+	defer r.Unlock()
+
+	other.Lock()
+	defer other.Unlock()
```

Comment:
This looks like it's racey, could deadlock if we have two results merging at once